.github/workflows/rust-compile.yml

```diff
@@ -155,3 +155,88 @@ jobs:
         if: ${{ !matrix.skip-tests }}
         run: >
           cargo test --doc
+
+  test-experimental:
```

Comment:
Ah I think I misunderstood what you meant initially. I think we should just enable the experimental feature for the regular tests. We dont need a whole new job.